packages/react/src/types.ts

```diff
@@ -53,6 +53,14 @@ export type UseUploadthingProps<
   TFileRoute extends AnyFileRoute,
   TServerOutput = inferEndpointOutput<TFileRoute>,
 > = {
+  /**
+   * Called when a file upload is completed
+   */
+  onFileUploadComplete?: (
+    uploadResponse: any,
+    index: number,
+    total: number,
+  ) => void;
```

Comment:
Not a fan of this API. I'm currently blocked on making changes to the progress callback since we chose to use positional arguments there instead of an object 😓 

```suggestion
  onFileUploadComplete?: (_: {
    uploadResponse: any, // <-- also, this shouldn't be any
    index: number,
    total: number,
  }) => void;
```